src/bun.js/webcore/response.zig

```diff
@@ -957,8 +1121,35 @@ pub const Headers = struct {
         else
             "";
     }
+    pub fn fromMerge(headers_ref: *FetchHeaders, b_headers_ref: *FetchHeaders, allocator: std.mem.Allocator) !Headers {
```

Comment:
This would result in duplicate HTTP headers being sent

Instead, we should probably make a C++ binding that creates a new headers object and merges an existing one (there is code for that, `fill` is the function name IIRC) 

Then we would call `from` on the merged FetchHeaders and deref it

This will be slower but it's okay because its the uncommon case

Reply:
make sense, i will take a look on that

Reply:
@Jarred-Sumner I tested node.js 18.9.1, chrome 108.0.5359.124 and deno 1.29.41.29.4

code :
```javascript
(async ()=> {
    const headers_a = new Headers();
    headers_a.append("Header-Test-1", "A");
    headers_a.append("Header-Test-2", "A");
    headers_a.append("Header-Test-3", "A");

    const headers_b = new Headers();
    headers_b.append("Header-Test-2", "B");
    headers_b.append("Header-Test-3", "B");
    headers_b.append("Header-Test-4", "B");

    const request = new Request("http://httpbin.org/get",{
        headers: headers_a
    });

    const result = await fetch(request, { headers: headers_b }).then((r) => r.text());
    console.log(result);
})();
```

node results:

```bash
{
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "gzip, deflate", 
    "Accept-Language": "*", 
    "Header-Test-2": "B", 
    "Header-Test-3": "B", 
    "Header-Test-4": "B", 
    "Host": "httpbin.org", 
    "Sec-Fetch-Mode": "cors", 
    "User-Agent": "undici", 
    "X-Amzn-Trace-Id": "Root=1-63cbedaf-65a061416ee27e87014bd575"
  }
}
```

chrome results:

```bash
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "gzip, deflate", 
    "Accept-Language": "pt-BR,pt;q=0.9,en-US;q=0.8,en;q=0.7", 
    "Cache-Control": "no-cache", 
    "Header-Test-2": "B", 
    "Header-Test-3": "B", 
    "Header-Test-4": "B", 
    "Host": "httpbin.org", 
    "Pragma": "no-cache", 
    "Referer": "http://httpbin.org/headers", 
    "User-Agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36", 
    "X-Amzn-Trace-Id": "Root=1-63cbede1-7fab3262249f4441783b050f"
  }, 
  "origin": "177.81.77.132", 
  "url": "http://httpbin.org/get"
}
```

deno results:

```bash
{
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "gzip, br", 
    "Accept-Language": "*", 
    "Header-Test-2": "B", 
    "Header-Test-3": "B", 
    "Header-Test-4": "B", 
    "Host": "httpbin.org", 
    "User-Agent": "Deno/1.29.4", 
    "X-Amzn-Trace-Id": "Root=1-63cbef0d-14608560361367cf3b6c8b10"
  }
}
```
In all cases, the headers are overwritten not merged. Should we follow this?

Reply:
Oh if it's supposed to overwrite then let's do that